### PR TITLE
[SPARK-25286][CORE] Removing the dangerous parmap

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

I propose to remove one of `parmap` methods which accepts an execution context as a parameter. The method should be removed to eliminate any deadlocks that can occur if `parmap` is called recursively on thread pools restricted by size.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@zsxwing @gatorsmile Please, have a look at the PR.

File: streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala [hunk 1]
```diff
@@ -315,7 +315,9 @@ private[streaming] object FileBasedWriteAheadLog {
     implicit val ec = executionContext
```
Comment (zsxwing) [inline streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala]: nit: this line is not needed

**Review by zsxwing — COMMENTED** [review]
LGTM except one nit

**Review by gatorsmile — COMMENTED** [review]
LGTM

**Comment by kiszk** [comment]
For future, let us keep a connection with #22233 that is the original one

**Comment by MaxGekk** [comment]
jenkins, retest this, please

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**